client/data/plugins/use-monitor-settings-mutation.ts

```diff
@@ -0,0 +1,65 @@
+import { UseQueryResult, useMutation, useQuery } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { SiteSlug } from 'calypso/types';
+
+export type UpdateMonitorURLOptions = {
+	status_down_webhook_url: string;
+};
+
+export type UpdateMonitorURL = {
+	monitor_url: string;
+	check_interval: number;
+	options: UpdateMonitorURLOptions;
+};
+
+export type UpdateMonitorSettings = {
+	wp_note_notifications?: boolean;
+	email_notifications?: boolean;
+	sms_notifications?: boolean;
+	jetmon_defer_status_down_minutes?: number;
+	urls?: UpdateMonitorURL[];
+};
+
+export type UpdateMonitorSettingsCreate = {
+	success: boolean;
+	settings: UpdateMonitorSettings;
+};
+
+export const useMonitorSettingsQuery = (
```

Comment:
It looks like this is not used. Is that for a future iteration? To prevent sending the create request on every new schedule?

Reply:
Yes, I thought we might need to read the preexisting, already-created monitors. I've seen no example of monitors that do not query the home page, but maybe there are some. I am not sure.